src/zcash/Proof.cpp

```diff
@@ -0,0 +1,185 @@
+#include "Proof.hpp"
+
+#include <boost/static_assert.hpp>
+
+#include "crypto/common.h"
+#include "libsnark/common/default_types/r1cs_ppzksnark_pp.hpp"
+#include "libsnark/zk_proof_systems/ppzksnark/r1cs_ppzksnark/r1cs_ppzksnark.hpp"
+
+using namespace libsnark;
+
+typedef default_r1cs_ppzksnark_pp curve_pp;
+typedef default_r1cs_ppzksnark_pp::G1_type curve_G1;
+typedef default_r1cs_ppzksnark_pp::G2_type curve_G2;
+typedef default_r1cs_ppzksnark_pp::GT_type curve_GT;
+typedef default_r1cs_ppzksnark_pp::Fp_type curve_Fr;
+typedef default_r1cs_ppzksnark_pp::Fq_type curve_Fq;
+typedef default_r1cs_ppzksnark_pp::Fqe_type curve_Fq2;
+
+namespace libzcash {
+
+BOOST_STATIC_ASSERT(sizeof(mp_limb_t) == 8);
+
+template<>
+Fq::Fq(curve_Fq element) : data()
+{
+    auto element_bigint = element.as_bigint();
+
+    unsigned char *p = data.begin();
```

Comment:
Is this guaranteed to have 4 limbs, or will it have only as many limbs as needed for the specific number?